src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -49,37 +52,47 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   /**
    * Create an event and add it to the calendar of the specified group. Query string should contain groupId
-   * POST body should contain: title,  description (opt), location (opt), start time, and end time.
+   * POST body should contain: title, description (opt), location (opt), start time, and duration (hours and mins).
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String groupId = getParameter(request, GROUP_ID_PROPERTY);
     String eventTitle = getParameter(request, EVENT_TITLE_PROPERTY);
     String eventStart = getParameter(request, EVENT_START_PROPERTY);
-    String eventEnd = getParameter(request, EVENT_END_PROPERTY);
+    String eventEnd = getEventEndTime(eventStart, getParameter(request, EVENT_HOURS_PROPERTY), getParameter(request, EVENT_MINUTES_PROPERTY));
 
     if (Strings.isNullOrEmpty(groupId) || Strings.isNullOrEmpty(eventTitle) 
-        || Strings.isNullOrEmpty(eventStart) || Strings.isNullOrEmpty(eventEnd)) {
+        || Strings.isNullOrEmpty(eventEnd)) {
       ServletUtils.printBadRequestError(response, EVENTS_POST_BAD_REQUEST_MESSAGE);
       return;
     }
 
     try {
       Event event = addEvent(ServletUtils.getGroupProperty(groupId, GROUP_CALENDARID_PROPERTY), 
           eventTitle, getParameter(request, EVENT_LOCATION_PROPERTY),
-          getParameter(request, EVENT_DESCRIPTION_PROPERTY), eventStart, eventEnd);
+          getParameter(request, EVENT_DESCRIPTION_PROPERTY), eventStart + TIMEZONE_OFFSET, eventEnd);
 
       response.setContentType(CONTENT_TYPE_JSON);
       response.getWriter().println(event.toString());
     } catch (Exception entityError) {
       response.getWriter().println(ENTITY_ERROR_MESSAGE);
     }
+    response.sendRedirect(CALENDAR_REDIRECT + groupId);
   }
 
   /**
-   * Called from another servlet to get the events rather than through a GET request
+   * Called from another servlet to get the events rather than through a GET request.
+   * Return null if invalid groupId.
    */
-  public Events getEventsList(String calendarId) throws IOException {
+  public Events getEventsList(String groupId) throws IOException {
+    String calendarId = "";
```

Comment:
is there a reason you set calendarId to be an empty string? does it give you an error if you declare it in the try block?

Reply:
yeah, since declaring it in the try/catch statement would make it out of scope for the return statement